juspay/hyperswitch
# refactor(required_fields): change required fields for billing address

## Type of Change



## Description

This PR changes the dynamic fields to collect `card_holder_name` as `payment_method_data.billing.first_name` and `payment_method_data.billing.last_name`. Backwards compatibility has been provided, so if `card_holder_name` is passed, then it will override the `payment_method_data.billing.first_name` and `payment_method_data.billing.last_name` passed in billing details.

All the required fields for billing details are changed to be collected as `payment_method_data.billing`

### Additional Changes


The billing details will now be collected in payment_method_billing.

## Motivation and Context

Standardise the way in which to collect billing details.

## How did you test it?

- Create a payment via sdk, check for connectors where `card_holder_name` is mandatory.
<img width="1122" alt="image" src="https://github.com/juspay/hyperswitch/assets/48803246/f406050b-313e-418d-a608-0b7d536a38e2">

- Check the /confirm call and confirm if card holder name is passed in billing details along with other billing fields.
<img width="1886" alt="image" src="https://github.com/juspay/hyperswitch/assets/48803246/795e8dd6-e504-4448-86df-a802fefc4472">

- Check whether required fields are rendered correctly by passing billing details in creating the payment and not passing billing details in creating the payment. When billing details are passed in payment create, then required fields should not be rendered. But when they are passed, required fields should be rendered in the sdk.

## Checklist
